binfmt/binfmt_execmodule.c

```diff
@@ -173,8 +174,15 @@ int exec_module(FAR const struct binary_s *binp,
       goto errout_with_envp;
     }
 
-  binfo("Initialize the user heap (heapsize=%d)\n", binp->addrenv.heapsize);
-  umm_initialize((FAR void *)CONFIG_ARCH_HEAP_VBASE, binp->addrenv.heapsize);
+  ret = up_addrenv_vheap(&binp->addrenv, &vheap);
+  if (ret < 0)
+    {
+      berr("ERROR: up_addrenv_vheap() failed: %d\n", ret);
+      goto errout_with_addrenv;
+    }
+
+  binfo("Initialize the user heap (heapsize=%zu)\n", binp->addrenv.heapsize);
```

Comment:
binp->addrenv.heapsize->up_addrenv_heapsize(&binp->addrenv). @pussuw could you create a new patch fix this?

Reply:
Sure